website/docs/migration/1.0.0_to_2.0.0.mdx

```diff
@@ -0,0 +1,246 @@
+
+This API ambiguity is solved with `Riverpod 2.0`.  
+Check the above example, migrated to its `Notifier` equivalent.
+<AutoSnippet language="dart" {...oldLifecycles}></AutoSnippet>
+
+Here, instead, `MyNotifier`, will *not* dispose when `durationProvider` updates:
+only its *internal state* will.  
+This breaking change simplifies the API and, hopefully, your DX, for two reasons:
+1. there is only one lifecycle to take care of, the one we really care about: `Notifier`'s inner state;
+2. there is only one API to register a `dispose` action: `ref.onDispose`.
+
+### There is no `.mounted` property
+WIP
+
+### Final remarks
+When transitioning to `Notifier` (or `AsyncNotifier`), it is good to keep in mind some more differences.
+
+#### Migrating from `.addListener` and `.stream`
+`StateNotifier`'s `.addListener` and `.stream` can be used to listen for state changes.
+These two APIs aren't a thing within `Notifier` or `AsyncNotifier`.
+
+This is intentional due to the API uniformity `Notifier` and `AsyncNotifier` offer.
+Indeed, using a `Notifier` or an `AsyncNotifier` shouldn't be any different from any other provider.
+
+So, if you want to listen to a `Notifier`, just use `ref.listen`, [as explained here].
+
+#### Migrating from `.debugState`
+`StateNotifier` exposed a `.debugState` getter that isn't a thing on `Notifier`.  
+This property had some use within `state_notifier`, as it enabled accessing state
+from outside the class when in development mode, aka for testing purposes.
+
+Again, for uniformity reasons, it doesn't make sense to expose such property on a `Notifier`
+(or on an `AsyncNotifier`).  
+If you want to interact with `Notifier` or `AsyncNotifier` in a testing environment, you should do
+it just like with any other provider: by reading them directly (as opposed of directly instantiating
+a `Notifier` by hand).
+
+This [testing guide] fully explains how to use `ProviderContainer.read` to do so.
+
+
+## Migrating from `StateProvider`
+
+`StateProvider` was exposed by Riverpod's libraries to save a few LoC when in need of a
+simplified version of `StateNotifierProvider`.  
+Since `StateNotifierProvider` is deprecated, `StateProvider` is to be avoided, too.
+
+To migrate from `StateProvider` and towards `Notifier` is simple. Take this example:
+
+```dart
+final counterProvider = StateProvider<int>((ref) {
+  return 0;
+});
+```
+
+When migrating to `Notifier`, this becomes:
+<AutoSnippet language="dart" {...fromStateProvider}></AutoSnippet>
+
+Even though it costs us a few more LoC, migrating away from `StateProvider` brings way more advantages, as shown above.
+
+## An improved `AsyncValue`
+WIP
+
+### `AsyncValue` can now expose two or more values at the same time
+e.g. "hasValue=true", "hasError=true" and "isLoading=true"
+WIP
+
+### How to work around the new APIs
+valueOrNull, hasError, hasData, asError, isLoading, etc.
+WIP
+
+## `ProviderListener` is not a thing anymore
```

Comment:
I don't think we need a full-on migration guide for this. The changelog should be fine